Feature/new markdown idevice

erseco opened the pull request:
## Summary

Adds a new **Markdown Text iDevice** that lets authors compose content in Markdown with a tabbed write/preview editor. The Markdown engine is the project's existing Showdown bundle, hardened a bit so that GitHub-style features and LaTeX both work end-to-end.

## What's in this PR

- New iDevice under `public/files/perm/idevices/base/markdown-text/` with edition (form + live preview) and export (runtime render + feedback toggle).
- `eXe.app.common.markdownToHTML` now enables Showdown's GFM options (tables, task lists, strikethrough, GFM-style sublist indentation) and protects LaTeX delimiters from being mangled by Markdown processing.
- `LIBRARY_PATTERNS.exe_highlighter` extended to also detect Showdown's `language-<lang>` classes, so Prism gets bundled in preview/export wherever fenced code blocks exist (not only TinyMCE-flagged blocks).
- Unit tests (vitest) for edition and export, plus a Playwright E2E spec covering add → edit → save → reload, preview tab, feedback rendering and attribute-escape behaviour.

## Improvements vs the initial version

- **GFM rendering**: tables, task-list checkboxes, strikethrough now render correctly.
- **Code blocks**: monospace fonts and `text-shadow: none` so dark-themed code doesn't look bold; Prism kicks in for fenced blocks in workarea, preview and export.
- **Lists**: ordered, unordered and nested lists work consistently.
- **LaTeX**: `\(...\)`, `\[...\]`, `$$...$$` and `\begin{}...\end{}` are preserved through the markdown pass and rendered by the existing MathJax integration in `renderBehaviour`.
- **Security**: user-supplied values that land inside HTML attributes (feedback button text, duration / participants labels and values) are HTML-escaped.
- **Bug fixes carried in**: LaTeX rendering no longer skipped when there's no feedback button, and the no-op mutation of the `data` argument inside `getHTMLView` (inherited from the text iDevice) is gone.

## How to test

Add the new "Markdown text" iDevice to a page, paste this content into the editor, and confirm the preview tab plus the saved view both render it as expected:

````markdown
# Main Title

## Subtitle

This is a paragraph with **bold text**, *italic text*, ***bold and italic text***, `inline code`, a [link](https://example.com), and an image:

![Random image from Picsum](https://picsum.photos/200/300)

---

## Lists

### Unordered List

- First item
- Second item
  - Nested item
  - Another nested item
- Third item

### Ordered List

1. First step
2. Second step
3. Third step

---

## Checkboxes

- [x] Create the repository
- [x] Add the README file
- [ ] Configure CI/CD
- [ ] Publish the first release

---

## Table

| Name | Role | Status | Priority |
|---|---|---:|:---:|
| Alice | Backend Developer | In progress | High |
| Bob | Frontend Developer | Pending | Medium |
| Carol | DevOps Engineer | Completed | Low |

---

## Quotes

> This is a quote.
>
> It can span multiple lines and highlight important ideas.

---

## Code Blocks

### Bash

```bash
#!/usr/bin/env bash

echo "Hello, world!"
```

### JSON

```json
{
  "name": "example-project",
  "version": "1.0.0",
  "private": true
}
```

### PHP

```php
<?php
/**
 * Return a greeting message.
 *
 * @param string $name User name.
 * @return string
 */
function get_greeting( string $name ): string {
    return sprintf( 'Hello, %s!', $name );
}
```

---

## Collapsible Details

<details>
<summary>Show more information</summary>

This content is hidden until the user expands the block.

- It can contain lists
- Code
- Tables
- Rich text

</details>

---

## Formula

A simple formula:

```text
total = price * quantity
```

LaTeX example for platforms that support it:

\(\cfrac{1}{1 + \cfrac{1}{2 + \cfrac{1}{3}}}\)

---

## Final Line

End of document.
````

## Test plan

- [x] `make fix && make test-unit && make test-integration` are green.
- [ ] `make test-e2e` (or at least `markdown-text.spec.ts` under chromium) passes.
- [ ] Manual smoke with the markdown above: tables render, task-list checkboxes appear, fenced code is colourised in workarea / preview / export, the LaTeX continued-fraction renders via MathJax, and saving + reloading the project preserves everything.

pabloamayab commented:
Great work, Ernesto. I have two questions:
- In order for the iDevice to appear, I had to do a make clean and delete volumes. We should test whether it would appear in the interface when installing a new version in an offline installation.
- I don't understand why task information appears, and I'm not even sure if the feedback area makes sense.
- 
<img width="1272" height="452" alt="image" src="https://github.com/user-attachments/assets/6c0e5cea-800a-468e-bf81-820a18f1d897" />

erseco commented:
Exactly.

The new *Markdown Text* iDevice inherits part of the structure and logic from the regular Text iDevice, which is why the task/feedback section appears. It’s not new behavior — it’s just a quick copy of that iDevice. For now, it can remains this way to keep compatibility with how eXe handles text-based activities, but we can review it later if it doesn’t make sense for Markdown.

Regarding the visibility issue — yes, you need to run `make clean` locally because the iDevice configuration files (`config.xml` and the cached iDevice JSON files) are stored in cache. In a clean installation or when volumes are rebuilt, it should appear automatically, even in offline mode.

We could also add a check or cleanup step to the build process to ensure it always loads correctly.

erseco committed:
feat(idevice): add Markdown Text iDevice with GFM and LaTeX support

Introduces a new "Markdown Text" iDevice that lets authors write content
in Markdown with a live write/preview tabbed editor.

Highlights:
- Showdown-powered rendering, with GitHub-flavored options enabled
  globally in app_common.markdownToHTML: tables, task lists,
  strikethrough, GFM-style sublist indentation.
- LaTeX delimiters (\(...\), \[...\], $$...$$, \begin{}...\end{})
  are stashed before Showdown runs and restored afterwards so that
  formulas survive the markdown pass and MathJax can pick them up at
  render time.
- Optional task-information block (duration, participants) and
  feedback section with a toggle button. User-supplied values are
  HTML-escaped where they land in attributes to avoid XSS.
- Syntax highlighting in workarea and exported HTML: the export
  library detector now recognises Showdown's "language-X" classes in
  addition to the legacy "highlighted-code", which loads exe_highlighter
  (Prism) wherever fenced code blocks appear.
- Registers the iDevice as componentType "json" in the browser-side
  shim so exe_export.js wires the feedback toggle in static exports
  and previews.
- Unit tests with vitest for both edition and export sides, plus a
  Playwright E2E spec covering add → edit → save → reload, preview tab,
  feedback in preview (attribute escape) and the feedback toggle click.

pabloamayab reviewed (approved):
LGTM!

ignaciogros committed:
Merge branch 'main' into feature/new-markdown-idevice
